.github/workflows/ci.yml

```
@@ -7,8 +7,41 @@ on:
+          node-version: "20"
+          cache: "npm"
+          cache-dependency-path: "server/package-lock.json"
+
+      - name: 📦 Install backend dependencies
+        run: npm ci
+
+      - name: 🧪 Run backend tests with coverage
+        run: npm run test:coverage:report
+
+      - name: 📊 Upload backend coverage reports
+        uses: codecov/codecov-action@v3
```

actionlint (1.7.8)

the runner of "codecov/codecov-action@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)